### PR TITLE
[TE] moved timerange persistence into services, set up acceptance test

### PR DESCRIPTION
**Pull request by newsummit**
**This covers**:
- Enable right-click on Alert search page
- Move time-range duration caching from session store to Ember services
- Set up acceptance test for time range persistence.

screen shot 2018-04-27 at 1 47 46 pm

**Context**:
The expected behavior in ThirdEye Self-serve is consistency of time ranges within alert management routes (under the same alert ID). Once the user clicks into a new alert, we reset any pre-selected time ranges to default (currently 3 months)

**Tests**:
312 tests completed in 37925 milliseconds, with 0 failed, 0 skipped, and 0 todo.
428 assertions of 428 passed, 0 failed.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js [hunk 1]
```diff
@@ -49,6 +49,11 @@ export default Controller.extend({
     resolution: 'anomalyFeedback'
   },
 
+  /**
+   * Make duration service accessible
+   */
+  durationCache: service('services/duration'),
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js]: Can we start with the convention custom services we built ends with `Service` and `ApiService` if it's for api?

example
```
  anomaliesApiService: service('services/api/anomalies'),
```

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  durationObj: null,
+  durationTypes: {
+    duration: 'string',
+    startDate: 'number',
+    endDate: 'number'
+  },
+
+  init() {
+    this._super(...arguments);
+    this.set('durationObj', {});
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: If you doing this. You can remove the declaration `durationObj: null,` i believe.

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   * @return {undefined}
+   */
+  setDuration(newDuration) {
+    const propsObj = this.get('durationTypes');
+    const requiredKeys = Object.keys(propsObj);
+    // Check each required property for presence and type
+    requiredKeys.forEach((key) => {
+      assert(`you must pass ${key} param as ${propsObj[key]}.`, typeof newDuration[key] === propsObj[key]);
+    });
+    this.set('durationObj', newDuration);
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: I don't understand this loop. Why are we setting the same value `newDuration` to the `durationObj` object?

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    return newDurationObj;
+  },
+
+  /**
+   * Clears the cached duration object (reset)
+   * @method removeDuration
+   * @return {undefined}
+   */
+  removeDuration() {
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: NIT: maybe `clearDuration` or `resetDuration` is better name?

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js (hunk 2 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: Yes - makes sense. I noticed we're doing it [here in RCA](https://github.com/linkedin/pinot/blob/b40464de8752453ac61ba5a69723977fae8643f0/thirdeye/thirdeye-frontend/app/pods/services/rootcause-aggregates-cache/service.js#L22) as well.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/controller.js]: Sounds good. However, this one is not related to API - no fetching of data from server. It only caches user-selected time ranges for various routes to read/update.

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js (hunk 3 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: Sure. `newDuration` is the incoming object containing the time-range properties to save.

It is not part of the loop. In the loop, I'm iterating over required keys to assert their presence and expected type.

File: thirdeye/thirdeye-frontend/app/pods/services/duration/service.js (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/services/duration/service.js]: Yes - good call.

**Pull request merged**